src/Microsoft.Azure.WebJobs.Extensions.Kafka/Config/KafkaExtensionConfigProvider.cs

```diff
@@ -71,15 +69,23 @@ public void Initialize(ExtensionConfigContext context)
 
         private IAsyncCollector<KafkaEventData> BuildCollectorFromAttribute(KafkaAttribute attribute)
         {
-            return new KafkaAsyncCollector(attribute.Topic, this.kafkaProducerManager.Get(attribute));
+            return new KafkaAsyncCollector(attribute.Topic, kafkaProducerManager.Get(attribute));
         }
 
         private ISpecificRecord ConvertKafkaEventData2AvroSpecific(KafkaEventData kafkaEventData)
         {
-            return (ISpecificRecord)kafkaEventData.Value;
+            if (kafkaEventData.Value is ISpecificRecord specRecord)
+            {
+                return specRecord;
+            }
+            else
+            {
+                logger.LogWarning($@"Unable to convert incoming data to Avro format. Expected ISpecificRecord, got {kafkaEventData.Value.GetType()}. Returning [null]");
```

Comment:
I didn't do error because previously we were simply returning `null` and not throwing an exception. So I warn, but let the `null` continue on through.